### DISCUSSION TITLE
Are there any more properties in listed ports?

### DISCUSSION THREAD

**Discussion by huagaojian** [serialport/node-serialport#2372] [Q&A] [answered]
SerialPort.list() returns the information:
```
[
  {
    path: 'COM19',
    manufacturer: 'wch.cn',
    serialNumber: undefined,
    pnpId: 'USB2MCOM\\USBE018_COM_01\\8&2587A489&0&0001',
    locationId: undefined,
    vendorId: undefined,
    productId: undefined
  },
  {
    path: 'COM8',
    manufacturer: 'Silicon Labs',
    serialNumber: '0242A1B6',
    pnpId: 'USB\\VID_10C4&PID_EA60\\0242A1B6',
    locationId: 'Port_#0001.Hub_#0005',
    vendorId: '10C4',
    productId: 'EA60'
  }
]
```

The properties of a port, only have: path, manufacturer, serialNumber, pnpId, locationId, vendorId, productId.
but on windows "Device Manager", we can see more properties of a port.

![903754c19adedd70c1c9c4cfdd89bfb](https://user-images.githubusercontent.com/22761684/147069249-a3ba354c-f8b7-4b3c-b813-ac98f1c437b4.png)

**How can I get these properties in listed information?**

**Comment by huagaojian** [comment, marked as answer]
From `https://github.com/serialport/electron-serialport` , I find it is called **friendlyName**, on SerialPorts version 10.x.x

**Reply by GazHank** [reply]
Oops, I think friendlyName was added into v10 (NAPI) when it was in early alpha testing (based on request https://github.com/serialport/node-serialport/pull/2329).  When that PR was declined, I must have forgotten to remove it from the alpha branch, and so it has made it's way into the live branch.

If you are actively using the friendlyName property, and would like to retain it then please let us know :-)

**Reply by gcampbell-msft** [reply]
Hi! We are actively using the friendlyName property and would love to retain it in the implementation of node-serialport if possible. Thanks!

**Reply by huagaojian** [reply]
Yes we are actively using the friendlyName property. by this feature, multi-channel serial driver chip can be clearly identified.
As the CHx channels shown below
![c51f7295f581d337b41e95bf3f79fbe](https://user-images.githubusercontent.com/22761684/176812087-1bef7af8-3b17-4c37-81b9-88c06868aee4.png)
